frontend/src/metabase/collections/containers/CollectionContent.jsx

```diff
@@ -363,20 +343,75 @@ function CollectionContent({
   );
 }
 
-export default _.compose(
-  Bookmark.loadList(),
-  Databases.loadList(),
-  Collection.loadList({
-    query: {
-      tree: true,
-      "exclude-other-user-collections": true,
-      "exclude-archived": true,
+export function CollectionContent({ collectionId }) {
+  const { data: bookmarks, error: bookmarksError } = useBookmarkListQuery();
+  const { data: databases, error: databasesError } = useDatabaseListQuery();
+  const { data: collections, error: collectionsError } = useCollectionListQuery(
+    {
+      query: {
+        tree: true,
+        "exclude-other-user-collections": true,
+        "exclude-archived": true,
+      },
     },
-    loadingAndErrorWrapper: false,
-  }),
-  Collection.load({
-    id: (_, props) => props.collectionId,
-    reload: true,
-  }),
-  connect(mapStateToProps, mapDispatchToProps),
-)(CollectionContent);
+  );
+  const { data: collection, error: collectionError } = useCollectionQuery({
+    id: collectionId,
+  });
+
+  const uploadDbId = useSelector(state =>
+    getSetting(state, "uploads-database-id"),
+  );
+  const uploadsEnabled = useSelector(state =>
+    getSetting(state, "uploads-enabled"),
+  );
+
+  const canUploadToDb = useSelector(
+    state =>
+      uploadDbId &&
+      Databases.selectors
+        .getObject(state, {
+          entityId: uploadDbId,
+        })
+        ?.canUpload(),
+  );
+
+  const isAdmin = useSelector(getUserIsAdmin);
+  const isNavbarOpen = useSelector(getIsNavbarOpen);
+
+  const dispatch = useDispatch();
+
+  const createBookmark = (id, type) =>
+    dispatch(Bookmark.actions.create({ id, type }));
+  const deleteBookmark = (id, type) =>
+    dispatch(Bookmark.actions.delete({ id, type }));
+
+  const uploadFile = dispatch(uploadFileAction);
+
+  if (!bookmarks || !databases || !collections || !collection) {
+    return <LoadingAndErrorWrapper loading />;
+  }
+
+  const error =
+    bookmarksError || databasesError || collectionsError || collectionError;
+  if (error) {
+    return <LoadingAndErrorWrapper error={error} />;
+  }
```

Comment:
if you get an error but no data, won't we keep showing the loading state? might be worth destructing the loading state off of all the `useXQuery` methods above.

Reply:
we do a good bit of reloading data in these components, so we don't want to be showing a loading spinner every time it goes back into loading state. I moved the error check above the loading check to make sure we hit that and don't end up with forever-loading states.